lib/public_key/src/pubkey_cert.erl

```diff
@@ -676,7 +676,15 @@ is_dir_name2({utf8String, Value1}, String) ->
     is_dir_name2({printableString, unicode:characters_to_list(Value1)}, String);
 is_dir_name2(String, {utf8String, Value1}) ->
     is_dir_name2(String, {printableString, unicode:characters_to_list(Value1)});
-is_dir_name2(_, _) ->
+is_dir_name2(Value1, Value2) when is_list(Value1), is_list(Value2) ->
+    Comp1 = string:to_lower(strip_spaces(Value1)),
```

Comment:
Why not use string:equal/3  with Ignorecase = true, instead of an obsolete to_lower function.
Using equal/3 you don't have to convert the binary to list either.

Reply:
If strip space is needed you should use string:casefold/1 instead of to_lower.

Reply:
Sure this code is pretty old so I will look into improving it according to your suggestion :) We probably want to release it on OTP-24 but that should be fine?!  public_key code is older than that.

Reply:
Should be good now with our common efforts :)